packages/app/src/systems/DApp/machines/transactionRequestMachine.tsx

```diff
@@ -251,19 +256,81 @@ export const transactionRequestMachine = createMachine(
+                transactionRequest: ctx.response?.proposedTxRequest,
+              };
+            },
           },
           onDone: [
             {
               target: 'failed',
-              actions: ['assignTxApproveError'],
+              actions: [
+                assign((context, event) => {
+                  return {
+                    errors: {
+                      ...(context.errors || {}),
+                      txApproveError: event.data as VMApiError,
+                    },
+                  };
+                }),
+              ],
               cond: FetchMachine.hasError,
             },
             {
-              actions: ['assignApprovedTx'],
+              actions: [
+                assign({
+                  response: (ctx, ev) => {
+                    const data = ev.data as MachineServices['send']['data'];
+
+                    const newResponseProperties: Partial<
+                      MachineContext['response']
+                    > = {};
+
+                    if (
+                      typeof data === 'object' &&
+                      data !== null &&
+                      'signedTransaction' in data &&
+                      typeof data.signedTransaction === 'string'
+                    ) {
+                      newResponseProperties.signedTransaction =
+                        data.signedTransaction;
+                      newResponseProperties.txResponse = undefined;
+                      newResponseProperties.txSummaryExecuted = undefined;
+                    } else if (
+                      typeof data === 'object' &&
+                      data !== null &&
+                      'gqlConnection' in data &&
+                      'id' in data &&
+                      typeof data.id === 'string'
+                    ) {
+                      // This is a TransactionResponse
+                      newResponseProperties.txResponse =
+                        data as TransactionResponse;
+                      newResponseProperties.signedTransaction = undefined;
+                      newResponseProperties.txSummaryExecuted = undefined;
+                    } else if (
+                      typeof data === 'object' &&
+                      data !== null &&
+                      'status' in data &&
+                      'id' in data &&
+                      typeof data.id === 'string'
+                    ) {
+                      // This should be a TransactionSummary
+                      newResponseProperties.txSummaryExecuted =
+                        data as TransactionSummary;
+                      newResponseProperties.txResponse = undefined;
+                      newResponseProperties.signedTransaction = undefined;
+                    }
+
+                    return {
+                      ...(ctx.response || {}),
+                      ...newResponseProperties,
+                    };
+                  },
+                }),
+              ],
```

Comment:
why did we need to go from a function with 10 lines to a function with 50 lines? can we keep the simple approach?